pkg/manifest/manifest.go

```diff
@@ -98,6 +110,69 @@ func (m *Manifest) UnmarshalJSON(in []byte) error {
 	return validateResourceId(m.id)
 }
 
+func (m *Manifest) Include(excludeIdentifier string, includeTechPreview bool, profile string, capabilities ClusterCapabilities) error {
+
+	annotations := m.Obj.GetAnnotations()
+	if annotations == nil {
+		return fmt.Errorf("no annotations")
+	}
+
+	excludeAnnotation := fmt.Sprintf("exclude.release.openshift.io/%s", excludeIdentifier)
+	if v := annotations[excludeAnnotation]; v == "true" {
+		return fmt.Errorf("%s=%s", excludeAnnotation, v)
+	}
+
+	featureGateAnnotation := "release.openshift.io/feature-gate"
+	featureGateAnnotationValue, featureGateAnnotationExists := annotations[featureGateAnnotation]
+	if featureGateAnnotationValue == "TechPreviewNoUpgrade" && !includeTechPreview {
+		return fmt.Errorf("tech-preview excluded, and %s=%s", featureGateAnnotation, featureGateAnnotationValue)
+	}
+	// never include the manifest if the feature-gate annotation is outside of allowed values (only TechPreviewNoUpgrade is currently allowed)
+	if featureGateAnnotationExists && featureGateAnnotationValue != "TechPreviewNoUpgrade" {
```

Comment:
Worth using `string(configv1.TechPreviewNoUpgrade)` here, to avoid the local string?

Reply:
Agree. Will fix it

Reply:
Fixed in https://github.com/openshift/library-go/commit/d2d9b0ce7cccf99dfd5903d246b0e619dc764096